src/lib/util/cdrom.h

```diff
@@ -58,6 +58,22 @@ enum
 #define CD_FLAG_GDROM   0x00000001  // disc is a GD-ROM, all tracks should be stored with GD-ROM metadata
 #define CD_FLAG_GDROMLE 0x00000002  // legacy GD-ROM, with little-endian CDDA data
 
+enum gdrom_area
+{
+	GDROM_SINGLE_DENSITY = 0,
+	GDROM_HIGH_DENSITY
+};
+
+enum gdrom_pattern
+{
+	GDROM_TYPE_UNKNOWN = 0,
+	GDROM_TYPE_I,
+	GDROM_TYPE_II,
+	GDROM_TYPE_III,
+	GDROM_TYPE_III_SPLIT
+};
```

Comment:
As I said before, this is an artificial classification based on observation of pressed discs, not something inherent to the format.  It doesn’t belong here.

Reply:
I've split the GDROM routines out to gdrom.h and gdrom.cpp and updated build files. 38a7fd72a15494d96bc228c7bfd5e6da5e79e357

Reply:
No, it doesn’t belong in a public header at all.  It should be internal to the CHD conversion code.  It’s an artificial classification and not something inherent to the disk format.

Reply:
It's used in two source files chdcd.cpp and chdman.cpp. It could go in chdcd.h although that's also a public header. If no header is acceptable what's the alternative?

Reply:
Moved the routines to chdcd.cpp in 0d4bb491024b8a3d92d27418b51b2e75a17fded0. Because the routines are used in chdman.cpp:do_extract_cd() the prototypes and enums need to be in chdcd.h. If that's not acceptable there is a more disruptive option that involves moving the bulk of do_extract_cd from chdman.cpp to chdcd.cpp. However that will affect existing code so the least-disruptive option is on the table first.